BasiliskII/src/CrossPlatform/sigsegv.cpp

```diff
@@ -342,12 +342,12 @@ static void powerpc_decode_instruction(instruction_t *instruction, unsigned int
 #define SIGSEGV_REGISTER_FILE			(unsigned long *)&SIGSEGV_CONTEXT_REGS->nip, (unsigned long *)(SIGSEGV_CONTEXT_REGS->gpr)
 #define SIGSEGV_SKIP_INSTRUCTION		powerpc_skip_instruction
 #elif (defined(arm) || defined(__arm__))
-#define SIGSEGV_CONTEXT_REGS			(((struct ucontext *)scp)->uc_mcontext)
+#define SIGSEGV_CONTEXT_REGS			(((ucontext_t *)scp)->uc_mcontext)
```

Comment:
Yeah, it should be safe on both platforms. I've just checked glibc git and on both arm and mips the `ucontext_t` typedef has been there since at least 1998:

https://sourceware.org/git/?p=glibc.git;a=blame;f=sysdeps/arm/sys/ucontext.h;hb=HEAD
https://sourceware.org/git/?p=glibc.git;a=blame;f=sysdeps/mips/sys/ucontext.h;hb=HEAD